gtsfm/frontend/two_view_ba_utils.py

```diff
@@ -0,0 +1,131 @@
+"""Functions to run Bundle Adjustment on a pair of images.
+
+Authors: Ayush Baid, John Lambert.
+"""
+import timeit
+from typing import Dict, List, Optional, Tuple
+
+import numpy as np
+from gtsam import Pose3, Rot3, SfmTrack, Unit3
+
+import gtsfm.common.types as gtsfm_types
+import gtsfm.utils.logger as logger_utils
+from gtsfm.bundle.bundle_adjustment import BundleAdjustmentOptimizer
+from gtsfm.common.gtsfm_data import GtsfmData
+from gtsfm.common.keypoints import Keypoints
+from gtsfm.common.pose_prior import PosePrior
+from gtsfm.common.sfm_track import SfmMeasurement, SfmTrack2d
+from gtsfm.data_association.point3d_initializer import Point3dInitializer, TriangulationOptions
+
+logger = logger_utils.get_logger()
+
+
+def triangulate_two_view_correspondences(
+    triangulation_options: TriangulationOptions,
+    cameras: Dict[int, gtsfm_types.CAMERA_TYPE],
+    keypoints_i1: Keypoints,
+    keypoints_i2: Keypoints,
+    corr_ind: np.ndarray,
+) -> Tuple[List[int], List[SfmTrack]]:
+    """Triangulate 2-view correspondences to form 3D tracks.
+
+    Args:
+        camera_i1: Camera for 1st view.
+        camera_i2: Camera for 2nd view.
+        keypoints_i1: Keypoints for 1st view.
+        keypoints_i2: Keypoints for 2nd view.
+        corr_ind: Indices of corresponding keypoints.
+
+    Returns:
+        Indices of successfully triangulated tracks.
+        Triangulated 3D points as tracks.
+    """
+    assert len(cameras) == 2
+    point3d_initializer = Point3dInitializer(cameras, triangulation_options)
+    triangulated_indices: List[int] = []
+    triangulated_tracks: List[SfmTrack] = []
+    for j, (idx1, idx2) in enumerate(corr_ind):
+        track2d = SfmTrack2d(
+            [SfmMeasurement(0, keypoints_i1.coordinates[idx1]), SfmMeasurement(1, keypoints_i2.coordinates[idx2])]
+        )
+        track, _, _ = point3d_initializer.triangulate(track2d)
+        if track is not None:
+            triangulated_indices.append(j)
+            triangulated_tracks.append(track)
+
+    return triangulated_indices, triangulated_tracks
+
+
+def bundle_adjust(
```

Comment:
Thanks for the PR @ayushbaid. QQ -- don't we already have code for 2-view BA here? https://github.com/borglab/gtsfm/blob/master/gtsfm/two_view_estimator.py#L123

Didn't quite follow the difference with the code here

Reply:
Oh I think I see now, we are just moving the other code to this new file.

Reply:
Yes. It helps me reuse it here. Although I am wondering if I should just do triangulation in correspondence augmenter and skip BA.